### PR TITLE
Add S7 options to get DB 'actual values' and expand arrays

### PR DESCRIPTION
**Pull request by ryan2445**
- adds options to get DB actual values and expand arrays for DBs
- assigns StartValue to expanded array items
- handling for getting actual values from more S7 data types

### CODE REVIEW

File: LibNoDaveConnectionLibrary/DataTypes/Projectfolders/Step7V5/BlocksOfflineFolder.cs [hunk 1]
```diff
@@ -423,14 +423,14 @@ private TmpBlock GetBlockBytes(string blkName)
            return GetBlockBytes(blkInfo);
         }
 
-        public S7DataRow GetInterface(string blkName)
+        public S7DataRow GetInterface(string blkName, S7ConvertingOptions myConvOpt)
```
Comment (jogibear9988) [inline LibNoDaveConnectionLibrary/DataTypes/Projectfolders/Step7V5/BlocksOfflineFolder.cs:426]: could you a here a overload without the Converting Options? So it is a non breaking change?

File: LibNoDaveConnectionLibrary/PLCs/S7_xxx/MC7/Helper.cs [hunk 2]
```diff
@@ -945,7 +945,7 @@ public static string GetFCPointer(byte b1, byte b2, byte b3, byte b4)
             switch (b1)
             {                
                 case 0x80:
-                    anf = "P#PE ";
+                    anf = "P#P ";
```
Comment (jogibear9988) [inline LibNoDaveConnectionLibrary/PLCs/S7_xxx/MC7/Helper.cs:948]: Is this correct? (I don't know what PE was... and what is P for...)

Comment (ryan2445) [reply LibNoDaveConnectionLibrary/PLCs/S7_xxx/MC7/Helper.cs:948]: I think so, at least that's how it looks in my software:
image

File: LibNoDaveConnectionLibrary/DataTypes/Projectfolders/Step7V5/BlocksOfflineFolder.cs (hunk 1 above)
Comment (ryan2445) [reply LibNoDaveConnectionLibrary/DataTypes/Projectfolders/Step7V5/BlocksOfflineFolder.cs:433]: yep!
https://github.com/dotnetprojects/DotNetSiemensPLCToolBoxLibrary/blob/169a959f549ed8fb6639134306ba2422b2545431/LibNoDaveConnectionLibrary/DataTypes/Projectfolders/Step7V5/BlocksOfflineFolder.cs#L426-L441

**Pull request merged**